src/main/java/net/dv8tion/jda/api/entities/Role.java

```diff
@@ -373,5 +373,47 @@ default String getIntegrationId()
         {
             return isIntegration() ? Long.toUnsignedString(getIntegrationIdLong()) : null;
         }
+
+        /**
+         * Whether this role can be acquired through a premium subscription purchase.
+         *
+         * @return True, if this is a subscription role
+         *
+         * @see    #getSubscriptionIdLong()
+         */
+        boolean hasSubscriptionListing();
+
+        /**
+         * The subscription listing id for this role.
+         *
+         * @return The listing id, or 0 if this role is not for a subscription listing
+         */
+        long getSubscriptionIdLong();
+
+        /**
+         * The subscription listing id for this role.
+         *
+         * @return The listing id, or null if this role is not for a subscription listing
+         */
+        @Nullable
+        default String getSubscriptionId()
+        {
+            return hasSubscriptionListing() ? Long.toUnsignedString(getSubscriptionIdLong()) : null;
+        }
+
+        /**
+         * Whether this role can be purchased by a user.
+         *
+         * @return True, if this role is purchasable
+         */
+        boolean isAvailableForPurchase();
+
+        /**
+         * Whether this role is acquired through a user connection.
+         * <br>Such as twitter or reddit.
+         *
+         * @return True, if this role is acquired through a user connection
+         */
+        boolean isGuildConnections();
```

Comment:
This.. naming isn't great.
does it not also apply for linked roles?

Reply:
I copied the name from the docs:

![image](https://user-images.githubusercontent.com/18090140/218278014-a2b590a4-d097-4be0-91d9-e3c91e91ec3c.png)

But I'm open for suggestions for better names. Maybe `isLinkedRole`?

Reply:
`isLinkedRole` is probably what we want here. I'm awaiting a response from an internal contact for more info.